### ISSUE TITLE
ShaderCompiler.py fails to parse shader

### ISSUE THREAD

**Issue by Shoshele** [y4my4my4m/kde-shader-wallpaper#44]
When trying to add a shader to the plasma6 version, running ShaderCompiler.py fails with the error message:
```
QSpirvCompiler: Failed to parse shader
Shader baking failed: ERROR: package/contents/ui/Shaders/ConvertMe/red bubbles 3.frag:3: 'ubuf' : undeclared identifier  
ERROR: package/contents/ui/Shaders/ConvertMe/red bubbles 3.frag:3: 'iResolution' : vector swizzle too long  
ERROR: package/contents/ui/Shaders/ConvertMe/red bubbles 3.frag:3: 'iResolution' : unknown swizzle selection  
ERROR: package/contents/ui/Shaders/ConvertMe/red bubbles 3.frag:3: 'iResolution' : unknown swizzle selection  
ERROR: package/contents/ui/Shaders/ConvertMe/red bubbles 3.frag:3: 'iResolution' : unknown swizzle selection  
ERROR: package/contents/ui/Shaders/ConvertMe/red bubbles 3.frag:3: '' : compilation terminated  
ERROR: 6 compilation errors.  No code generated.  
Conversion failed for: red bubbles 3.frag
```
The file, red bubbles 3.frag, is a simple shader that has worked flawlessly in the plasma5 version. I've attached the file (as a txt due to github limitations). I understand the issue is probably with my shader rather than with the compiler, I still don't fully understand how any of it works.

[bubbles.txt](https://github.com/y4my4my4m/kde-shader-wallpaper/files/14875866/bubbles.txt)

**Comment by y4my4my4m** [comment]
@Shoshele you need to run ShaderProcessor.py first.

Unfortunately the processor is very weak codewise and it probably will fail since you don't  have double newline at the top (I'll improve it, it's kinda rushed right now).

Read the README_Dev.md to add the relevant missing shader part at the top and beginning of the file.

**Comment by Shoshele** [comment]
I did run ShaderProcessor first, but still got the issue. But thank you for pointing me in the right direction! Adding the definitions and main() manually, and running qsb instead of ShaderCompiler, gave me compiler errors I actually could understand, and I was able to edit bubbles.frag such that it would successfully compile, and it's now running quite well under plasma6.

Am I correct to think the plasma6 requirement for compilation is why it no longer supports editing parts of the shader in the configuration? That's unfortunate.
